pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/ProtoWriterToolsTest.java

```diff
@@ -986,6 +1059,268 @@ void testSizeOfDelimited() {
+
+        final long start = bufferedData.position();
+        ProtoWriterTools.writeIntegerList(bufferedData, definition, List.of(
+                0x0f,
+                0xff,
+                0xfff,
+                0xffff,
+                0xfffff,
+                0xffffff,
+                0xfffffff,
+                0xffffffff
+        ));
+        final long finish = bufferedData.position();
+
+        int tag = bufferedData.getVarInt(start, false);
+        assertEquals(WIRE_TYPE_DELIMITED.ordinal(), tag & 0x3);
+        int sizeOfTag = ProtoWriterTools.sizeOfVarInt32(tag);
+
+        int size = bufferedData.getVarInt(start + sizeOfTag, false);
+        assertEquals(expectedSize, size);
+        int sizeOfSize = ProtoWriterTools.sizeOfVarInt32(size);
+
+        assertEquals(finish - start - sizeOfTag - sizeOfSize, size);
+
+        int firstInt = bufferedData.getVarInt(start + sizeOfTag + sizeOfSize, zigZag);
+        assertEquals(0x0f, firstInt);
+    }
+
+    private static Stream<Arguments> provideWriteLongListArguments() {
+        return Stream.of(
+                Arguments.of(INT64, 85, false),
+                Arguments.of(UINT64, 85, false),
+                Arguments.of(SINT64, 78, true),
+                Arguments.of(FIXED64, 128, false),
+                Arguments.of(SFIXED64, 128, false)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideWriteLongListArguments")
+    void testWriteLongList(final FieldType type, final int expectedSize, final boolean zigZag) {
+        final FieldDefinition definition = createRepeatedFieldDefinition(type);
+
+        final long start = bufferedData.position();
+        ProtoWriterTools.writeLongList(bufferedData, definition, List.of(
+                0x0fL,
+                0xffL,
+                0xfffL,
+                0xffffL,
+                0xfffffL,
+                0xffffffL,
+                0xfffffffL,
+                0xffffffffL,
+                0xfffffffffL,
+                0xffffffffffL,
+                0xfffffffffffL,
+                0xffffffffffffL,
+                0xfffffffffffffL,
+                0xffffffffffffffL,
+                0xfffffffffffffffL,
+                0xffffffffffffffffL
+        ));
+        final long finish = bufferedData.position();
+
+        int tag = bufferedData.getVarInt(start, false);
+        assertEquals(WIRE_TYPE_DELIMITED.ordinal(), tag & 0x3);
+        int sizeOfTag = ProtoWriterTools.sizeOfVarInt32(tag);
+
+        int size = bufferedData.getVarInt(start + sizeOfTag, false);
+        assertEquals(expectedSize, size);
+        int sizeOfSize = ProtoWriterTools.sizeOfVarInt32(size);
+
+        assertEquals(finish - start - sizeOfTag - sizeOfSize, size);
+
+        long firstLong = bufferedData.getVarLong(start + sizeOfTag + sizeOfSize, zigZag);
+        assertEquals(0x0f, firstLong);
+    }
+
+    private static interface WriterMethod<T> {
```

Comment:
I'm not sure, but can `ProtoWriter` interface be reused here?

Reply:
The ProtoWriter seems to accept an output sink and a single data variable, while this interface would be unable to infer a particular field definition from the given data object alone. So I need this custom interface here that also accepts the field definition.